### PR TITLE
Enable AOT compilation by default.

### PR DESCRIPTION
**Pull request by brendandahl**
I'm still working on benchmarking for a better list of methods to compile ahead of time.

### CODE REVIEW

**Comment by brendandahl** [comment]
I've updated the method list with one built from starting up and moving around in a midlet. It's hard to tell if it really helps that much. On desktop I can semi-regularly get <1s to the first refresh0 on the splash screen, whereas without I usually saw ~1.1s. Still working on tweaking it, but this requires finding a better way to benchmark to see how much we should compile AOT.

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ before_script:
   - wget -P /tmp/j2me.js -N https://ftp.mozilla.org/pub/mozilla.org/xulrunner/releases/31.0/runtimes/xulrunner-31.0.en-US.linux-x86_64.tar.bz2
   - tar x -C /tmp -f /tmp/j2me.js/xulrunner-31.0.en-US.linux-x86_64.tar.bz2
   - export SLIMERJSLAUNCHER=/tmp/xulrunner/xulrunner
+  - wget -P /tmp/j2me.js -N http://ftp.mozilla.org/pub/mozilla.org/firefox/nightly/36.0b1-candidates/build1/jsshell-linux-x86_64.zip
```
Comment (mykmelez) [inline .travis.yml]: Nit: this should use the https: URL for the file.

File: Makefile [hunk 2]
```diff
@@ -26,34 +26,35 @@ build/jsc.js: jsc.ts build/j2me-jsc.js
...
 	@echo "Compiling ..."
-	js build/jsc.js -cp java/classes.jar -d -jf java/classes.jar > build/classes.jar.js
-	js build/jsc.js -cp java/classes.jar tests/tests.jar -d -jf tests/tests.jar > build/tests.jar.js
-	if test -f program.jar; then \
-		js build/jsc.js -cp java/classes.jar program.jar -d -jf program.jar > build/program.jar.js; \
-	fi
-	@echo "Done"
-
-closure: build/j2me.js aot
+	js build/jsc.js -cp java/classes.jar -d -jf java/classes.jar -mff aot-methods.txt > build/classes.jar.js
```
Comment (mykmelez) [inline Makefile]: Nice refactoring; I like these separate [classes|tests|program].jar.js build targets much better.

File: aot-methods.txt [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+javax/microedition/lcdui/Image.<init>.(Ljavax/microedition/lcdui/ImageData;)V
+javax/microedition/lcdui/ImageData.<init>.(IIZ)V
+com/sun/midp/events/Event.<init>.(I)V
+javax/microedition/lcdui/Displayable.getHeight.()I
+javax/microedition/lcdui/DisplayableLFImpl.lGetHeight.()I
+javax/microedition/lcdui/Displayable.getWidth.()I
+javax/microedition/lcdui/DisplayableLFImpl.lGetWidth.()I
+java/lang/Object.wait.()V
+java/lang/Boolean.booleanValue.()Z
+com/whatsapp/client/test/ContactListMidlet.getInstance.()Lcom/whatsapp/client/test/ContactListMidlet;
```
Comment (mykmelez) [inline aot-methods.txt]: This class isn't part of classes.jar and thus shouldn't be in this list.

File: aot-methods.txt [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+javax/microedition/lcdui/ImageDataFactory.createResourceImageData.(Ljava/lang/String;)Ljavax/microedition/lcdui/ImageData;
+javax/microedition/lcdui/ImageData.<init>.()V
+javax/microedition/lcdui/ImageDataFactory.loadAndCreateImmutableImageDataFromCache.(Ljavax/microedition/lcdui/ImageData;Ljava/lang/String;)Z
+javax/microedition/lcdui/SuiteImageCacheImpl.loadAndCreateImmutableImageData.(Ljavax/microedition/lcdui/ImageData;Ljava/lang/String;)Z
+com/sun/midp/io/Util.toJavaString.([B)Ljava/lang/String;
+java/util/Hashtable$HashtableEnumerator.nextElement.()Ljava/lang/Object;
+com/sun/midp/lcdui/DisplayDevice.isPrimaryDisplay.()Z
+com/sun/midp/main/MIDletProxy.getClassName.()Ljava/lang/String;
+com/sun/midp/rms/RmsEnvironment.getSecureFilenameBase.(I)Ljava/lang/String;
+com/sun/midp/midletsuite/SuiteContainerAdapter.getSecureFilenameBase.(I)Ljava/lang/String;
```
Comment (mykmelez) [inline aot-methods.txt]: There are 351 methods in this list, but only 349 get compiled on my machine, although "0 failed":

> Compiling ...
> js build/jsc.js -cp java/classes.jar -d -jf java/classes.jar -mff aot-methods.txt > build/classes.jar.js
> Compiled 349 methods OK, 0 failed.

One of the two missing methods is presumably the one I mentioned earlier. It isn't clear what the other one is. But the compiler should warn when a method is missing, so we can prune the list accordingly.

**Comment by mykmelez** [comment]
Nit: there are a bunch of graphics classes in the list that aren't found anymore, probably because @TimAbraldes removed them:

```
Compiled 334 methods OK, 0 failed.
The following method(s) in the method filter list failed to compile or were not found:
 javax/microedition/lcdui/Graphics.getClipX.()I
 javax/microedition/lcdui/Graphics.getClipY.()I
 javax/microedition/lcdui/Graphics.getClipWidth.()I
 javax/microedition/lcdui/Graphics.getClipHeight.()I
 javax/microedition/lcdui/Graphics.setClip.(IIII)V
 javax/microedition/lcdui/Graphics.clipRect.(IIII)V
 javax/microedition/lcdui/Graphics.setColor.(I)V
 javax/microedition/lcdui/Graphics.getFont.()Ljavax/microedition/lcdui/Font;
 javax/microedition/lcdui/Graphics.getColor.()I
 javax/microedition/lcdui/Graphics.setFont.(Ljavax/microedition/lcdui/Font;)V
 javax/microedition/lcdui/Graphics.grayVal.(III)I
 javax/microedition/lcdui/Graphics.resetGC.()V
 javax/microedition/lcdui/Graphics.reset.(IIII)V
 javax/microedition/lcdui/Graphics.setDimensions.(II)V
 javax/microedition/lcdui/Graphics.translate.(II)V
```

**Comment by mykmelez** [comment]
Merged on command line with one last method that no longer exists (*com/nokia/mid/ui/TextEditor.getVisibleContentPosition.()I*) removed in 4775e270f214298659cee49037a16e4324b9c078.

**Pull request merged**